File: d10/D10System.swift

extension D10 {
  
  enum System {
    
    static let patchWerk = try! DXX.sysexWerk.singlePatchWerk("System", parms.params(), size: 0x32, start: 0x100000, initFile: "d10-system-init")

    static let parms: [Parm] = D110.System.parms + [
    ] + .prefix([.part], count: 8, bx: 1, block: { i, off in
      [
        .p([.level], 0x21, .max(100)),
        .p([.pan], 0x2a, .max(14, dispOff: -7)),
      ]
    }) + [
      .p([.part, .rhythm, .level], 0x29, .max(100)),
    ]

  }
  
}

File: d10/d10_system.js
const D110System = require('./d110_system.js')

const parms = D110System.parms.concat([
  { prefix: "part", count: 8, bx: 1, block: [
    ['level', { b: 0x21, max: 100 }],
    ['pan', { b: 0x2a, max: 14, dispOff: -7 }],
  ] },
  ['part/rhythm/level', { b: 0x29, max: 100 }],
])

const patchWerk = {
  single: "System",
  parms: parms,
  size: 0x32,
  initFile: "d10-system-init",
}

module.exports = {
  patchWerk,
}